File: src/contests/cg/puzzle/max-area.ts
// 🎮 CodinGame Puzzle - max-area
// https://www.codingame.com/training/easy/max-area

const n = parseInt(readline(), 10)
const a = readline()
  .split(" ")
  .map(x => parseInt(x, 10))
let lo = 0
let hi = n - 1
let best = 0
while (lo < hi) {
  const h = Math.min(a[lo], a[hi])
  const area = h * (hi - lo)
  if (area > best) best = area
  if (a[lo] < a[hi]) lo++
  else hi--
}
console.log(best)

File: src/contests/cg/puzzle/csharp/max-area.cs
// 🎮 CodinGame Puzzle - max-area
// https://www.codingame.com/training/easy/max-area

using System;

class Solution
{
    static void Main()
    {
        Console.ReadLine();
        var a = Array.ConvertAll(Console.ReadLine().Trim().Split(' '), long.Parse);
        int left = 0, right = a.Length - 1;
        long best = 0;
        while (left < right)
        {
            long h = Math.Min(a[left], a[right]);
            long area = h * (right - left);
            if (area > best) best = area;
            if (a[left] < a[right]) left++; else right--;
        }
        Console.WriteLine(best);
    }
}
